tests/integ/test_stored_procedure.py

```diff
@@ -78,13 +84,37 @@ def int2str(session_, x):
     assert pow_sp(2, 10, session=session) == 1024
 
 
+def test_stored_procedure_with_column_datatype(session):
+    def plus1(session_, x):
+        return x + 1
+
+    def add(session_, x, y):
+        return x + y
+
+    def add_date(session_, date, add_days):
+        return date + datetime.timedelta(days=add_days)
+
+    plus1_sp = sproc(plus1, return_type=IntegerType(), input_types=[IntegerType()])
+    add_sp = sproc(
+        add, return_type=IntegerType(), input_types=[IntegerType(), IntegerType()]
+    )
+    add_date_sp = sproc(
+        add_date, return_type=DateType(), input_types=[DateType(), IntegerType()]
+    )
+
+    dt = datetime.date.today() + datetime.timedelta(days=3)
+    assert plus1_sp(lit(6)) == 7
+    assert add_sp(4, sqrt(lit(36))) == 10
+    assert add_date_sp(current_date(), 3) == dt
```

Comment:
Let's add some negative cases? For example, `plus1_sp(col("a"))`

Reply:
cool. added some in the new commit